storage/blob.go

```diff
@@ -52,8 +87,40 @@ type Blobs interface {
 	Create(ctx context.Context, ref BlobRef, size int64) (BlobWriter, error)
 	// Open opens a reader with the specified namespace and key
 	Open(ctx context.Context, ref BlobRef) (BlobReader, error)
+	// OpenSpecific opens a reader for the already-located blob, avoiding the potential need
+	// to check multiple storage formats to find the blob.
+	OpenSpecific(ctx context.Context, ref BlobRef, formatVer FormatVersion) (BlobReader, error)
 	// Delete deletes the blob with the namespace and key
 	Delete(ctx context.Context, ref BlobRef) error
+	// Lookup looks up disk metadata on the blob file
+	Lookup(ctx context.Context, ref BlobRef) (StoredBlobAccess, error)
+	// LookupSpecific looks up disk metadata for the blob file with the given storage format
+	// version. This avoids the potential need to check multiple storage formats for the blob
+	// when the format is already known.
+	LookupSpecific(ctx context.Context, ref BlobRef, formatVer FormatVersion) (StoredBlobAccess, error)
 	// FreeSpace return how much free space left for writing
 	FreeSpace() (int64, error)
+	// SpaceUsed adds up how much is used in all namespaces
+	SpaceUsed(ctx context.Context) (int64, error)
+	// SpaceUsedInNamespace adds up how much is used in the given namespace
+	SpaceUsedInNamespace(ctx context.Context, namespace []byte) (int64, error)
+	// GetAllNamespaces finds all namespaces in which keys might currently be stored.
+	GetAllNamespaces(ctx context.Context) ([][]byte, error)
+	// ForAllKeysInNamespace executes doForEach for each locally stored blob, stored with
+	// storage format V1 or greater, in the given namespace. If doForEach returns a non-nil
+	// error, ForAllKeysInNamespace will stop iterating and return the error immediately.
+	ForAllKeysInNamespace(ctx context.Context, namespace []byte, doForEach func(StoredBlobAccess) error) error
```

Comment:
Maybe `WalkNamespace` and `doForEach` -> `walkFunc`?

Reply:
I suppose